src/local.py

```diff
@@ -40,6 +38,12 @@ def get_local_launcher_path(self):
             self.installer_location = None
         return self.installer_location
 
+    async def kill_launcher(self):
+        # The Launcher exits without displaying an error message if LauncherPatcher.exe is killed before Launcher.exe.
+        subprocess.Popen("taskkill /f /im LauncherPatcher.exe")
+        await asyncio.sleep(1)
+        subprocess.Popen("taskkill /f /im Launcher.exe")
```

Comment:
Killing ```Launcher.exe``` does seem to halt any cloud save uploads, but killing ```SocialClubHelper.exe``` seems to send the proper shutdown signal to the launcher. Here is the relevant snippet from the Rockstar Games Launcher log file after launching a game via Galaxy 2.0, closing it within the game, and then having the plugin send ```taskkill /im SocialClubHelper.exe``` via ```subprocess.Popen```:

```
[2019-11-26 13:04:33.928] [DISPLAY] [12988] [gamelaunch] Launching game...
[2019-11-26 13:04:33.928] [DISPLAY] [12988] [gamelaunch]  Path: C:\SSDGames\GOG Galaxy\Red Dead Redemption 2\RDR2.exe
[2019-11-26 13:04:33.928] [DISPLAY] [12988] [gamelaunch]  Command Line: "C:\SSDGames\GOG Galaxy\Red Dead Redemption 2\RDR2.exe" -minmodeapp=rdr2 -launchTitleInFolder "C:\SSDGames\GOG Galaxy\Red Dead Redemption 2" @commandline.txt -USEALLAVAILABLECORES -high
[2019-11-26 13:04:33.928] [DISPLAY] [12988] [gamelaunch]  Working Directory: C:\SSDGames\GOG Galaxy\Red Dead Redemption 2
[2019-11-26 13:05:21.653] [DISPLAY] [12988] [gamelaunch] Game exited with code 0x0 (0)
[2019-11-26 13:05:21.653] [DISPLAY] [12988] [crashdetection] Exit code 0x0 indicates a clean game shutdown
[2019-11-26 13:05:21.684] [DISPLAY] [Main ] [cloudsaveop] Cloud Save sync requested for title 'rdr2'
[2019-11-26 13:05:21.701] [DISPLAY] [Main ] [cloudsaveop] Starting cloud save enabled check for title 'rdr2'
[2019-11-26 13:05:21.704] [DISPLAY] [Main ] [cloudsaveop] Received cloud saves enabled state for title 'rdr2' = enabled
[2019-11-26 13:05:21.704] [DISPLAY] [Main ] [cloudsaveop] Identifying potential cloud save conflicts for title 'rdr2'
[2019-11-26 13:05:22.328] [DISPLAY] [Main ] [cloudsaveop] Received modification list of cloud save files for title 'rdr2'
[2019-11-26 13:05:22.328] [DISPLAY] [Main ] [cloudsaveop] Retrieving conflict metadata for title 'rdr2'
[2019-11-26 13:05:22.345] [DISPLAY] [Main ] [cloudsaveop] Downloaded conflict metadata for title 'rdr2'
[2019-11-26 13:05:22.345] [DISPLAY] [Main ] [cloudsaveop] Posting updated files for title 'rdr2'
[2019-11-26 13:05:22.362] [DISPLAY] [Main ] [cloudsaveop] File uploading completed for title 'rdr2'
[2019-11-26 13:05:22.362] [DISPLAY] [Main ] [cloudsaveop] Getting update files for title 'rdr2'
[2019-11-26 13:05:22.379] [DISPLAY] [Main ] [cloudsaveop] File downloading completed for title 'rdr2'
[2019-11-26 13:05:22.379] [DISPLAY] [Main ] [cloudsaveop] Deleting updated files for title 'rdr2'
[2019-11-26 13:05:22.396] [DISPLAY] [Main ] [cloudsaveop] Files deletion completed for title 'rdr2'
[2019-11-26 13:05:22.396] [DISPLAY] [Main ] [cloudsaveop] Swapping updated files in place for title 'rdr2'
[2019-11-26 13:05:22.416] [DISPLAY] [Main ] [cloudsaveop] Finished syncing cloud saves for title 'rdr2'
[2019-11-26 13:05:22.787] [DISPLAY] [Main ] [scuimessage] Sending version info to Social Club UI
[2019-11-26 13:05:22.789] [DISPLAY] [Main ] [external] Notifying Social Club UI about external title launch on awake...
[2019-11-26 13:05:22.789] [DISPLAY] [ 5136] [sysinfo] Drive C:\ has 28141641728 bytes free, 999559262208 bytes total
[2019-11-26 13:05:22.789] [DISPLAY] [Main ] [ui_mainwindow] Using DPI 119,119
[2019-11-26 13:05:22.789] [ ERROR ] [ 5136] [sysinfo] Failed to get free and total space on drive D:\!
[2019-11-26 13:05:22.789] [DISPLAY] [ 5136] [sysinfo] Drive F:\ has 2184969609216 bytes free, 3000574668800 bytes total
[2019-11-26 13:05:22.790] [ ERROR ] [ 5136] [sysinfo] Failed to get free and total space on drive G:\!
[2019-11-26 13:05:25.878] [DISPLAY] [Main ] [launcher] StartShutdown - Now in progress
[2019-11-26 13:05:26.448] [DISPLAY] [Main ] [app] Application quit requested (exit code: 0)
[2019-11-26 13:05:26.689] [DISPLAY] [Main ] [launcher] Closing log file.
```

Reply:
Cool, looks legit. Resolving.

In the case you shown 13:05:21 game is closed, cloud saves syncing lasted less than one second and then launcher started shutdown in 13:05:25. If SocialClubHelper.exe was killed along with the game process, its OK, otherwise you should test with bigger saves upload. It may lasts much longer then 1 sec.